dbms/src/Flash/Pipeline/WaitReactor.h

```diff
@@ -0,0 +1,53 @@
+// Copyright 2022 PingCAP, Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#include <Common/Logger.h>
+#include <Flash/Pipeline/Task.h>
+#include <Flash/Pipeline/WaitQueue.h>
+
+#include <list>
+#include <thread>
+
+namespace DB
+{
+class TaskScheduler;
+
+class WaitReactor
+{
+public:
+    explicit WaitReactor(TaskScheduler & scheduler_);
+
+    void close();
+
+    void waitForStop();
+
+    void submit(TaskPtr && task);
+
+    void submit(std::list<TaskPtr> & tasks);
+
+private:
+    void loop() noexcept;
```

Comment:
setThreadName() will throw exception, I am wondering if the `noexcept` safe

Reply:
if `setThreadName` throw Exception, the tiflash node crash is expected.

Reply:
Both Task and Event are expected not to report errors, and do try catch internally.
Anything else is an unintended error and should make tiflash crash.